### ISSUE TITLE
Add support to dump the shadow file

### ISSUE THREAD

**Issue by jannikuhl** [mattcabb/mib2std-toolbox#2]
Have you already had success reading the shadow file?

I am currently trying to access the MIB via Telnet. I can connect, but I don't have the root password.

Unfortunately I can't contact you anywhere. If you are interested in exchanging information I would be happy if you send me an email to jannik.uhlmann(at)icloud.com.

I am currently working on reverse engineering the FEC/SWaP system of Technisat.

**Comment by mattcabb** [comment]
Nothing yet. Due to lockdown I had to put most of things on hold for now. Hope to get back to the hack soon.

**Comment by flipidus** [comment]
> 
> 
> Have you already had success reading the shadow file?
> 
> I am currently trying to access the MIB via Telnet. I can connect, but I don't have the root password.
> 
> Unfortunately I can't contact you anywhere. If you are interested in exchanging information I would be happy if you send me an email to jannik.uhlmann(at)icloud.com.
> 
> I am currently working on reverse engineering the FEC/SWaP system of Technisat.

Hello Jannik,

im am also trying to get inside the MIB2STD and im building a testing station at home. I am also interested in exchanging some information about this topic. I heard there is also a serial connection in the quadlock connector but i dont know if this is only on a HIGH device or also on a STD device. What tools you use exactly to  establish a telnet connection?

Best regards

**Comment by jannikuhl** [comment]
> > Have you already had success reading the shadow file?
> > I am currently trying to access the MIB via Telnet. I can connect, but I don't have the root password.
> > Unfortunately I can't contact you anywhere. If you are interested in exchanging information I would be happy if you send me an email to jannik.uhlmann(at)icloud.com.
> > I am currently working on reverse engineering the FEC/SWaP system of Technisat.
> 
> Hello Jannik,
> 
> im am also trying to get inside the MIB2STD and im building a testing station at home. I am also interested in exchanging some information about this topic. I heard there is also a serial connection in the quadlock connector but i dont know if this is only on a HIGH device or also on a STD device. What tools you use exactly to establish a telnet connection?
> 
> Best regards

The MIB2STD does not have Telnet enabled by default and currently the only way to enable it is writing on the bench. Either by soldering or using (what I prefer) BDM. You need to add the following line to the file /fs/hd1-qnx6/tsd/bin/system/startup
`echo ser1 "/bin/login -f root" qansi-m on > /tmp/ttys/sbin/tinit -f /tmp/ttys &`

I never heard about a serial connection on the quadlock, do you have any sources?
It is also possible to inject patched binary files as a software update.

**Comment by mattcabb** [comment]
Found this in test mode.
![debug](https://user-images.githubusercontent.com/2262989/81163783-915b5a80-8f8f-11ea-9eab-48238878d8c0.png)

"Serial" might be related to the quadlock (B:J5_TX, B:J5_RX)
![mib2_wiring_b2edb871d27541030fbdd66275bc7a87993b16c3](https://user-images.githubusercontent.com/2262989/81163791-94eee180-8f8f-11ea-8939-178e8ae313b9.jpg)

**Comment by flipidus** [comment]
> 
> 
> Found this in test mode.
> ![debug](https://user-images.githubusercontent.com/2262989/81163783-915b5a80-8f8f-11ea-9eab-48238878d8c0.png)
> 
> "Serial" might be related to the quadlock (B:J5_TX, B:J5_RX)
> ![mib2_wiring_b2edb871d27541030fbdd66275bc7a87993b16c3](https://user-images.githubusercontent.com/2262989/81163791-94eee180-8f8f-11ea-8939-178e8ae313b9.jpg)

yes thats the serial connection i heard about it after a longer googeling about that stuff. But this PIN-Out in your picture is from a Label on a MIB2 HIGH Device, so i dont know if there is also a Serial Connection on a MIB2 STD device on this Pins 3 and 9. i also readed that you need pin 7 for the serial connection for the GND.

Do you know what the Pins 11 "ESO C3_TX" and 12 "ESO C3_RX" are on the A Part of the Quadlock Connector?

**Comment by jannikuhl** [comment]
Where have you found that setting? It looks like mine doesn't have that.
What units are you guys exactly have?

I have a Skoda Technisat MST2Nav unit.

We have to be careful because the MIB2STD unit with the same partnumber is manufactured by two different manufacturers (Technisat and Delphi). So maybe we have to split the toolbox since they both work very different.

I think the pinout is from a Porsche PCM 4.0 which is manufactured by Harman/Becker. These are both very different units.

**Comment by flipidus** [comment]
> 
> 
> Where have you found that setting? It looks like mine doesn't have that.
> What units are you guys exactly have?
> 
> I have a Skoda Technisat MST2Nav unit.
> 
> We have to be careful because the MIB2STD unit with the same partnumber is manufactured by two different manufacturers (Technisat and Delphi). So maybe we have to split the toolbox since they both work very different.
> 
> I think the pinout is from a Porsche PCM 4.0 which is manufactured by Harman/Becker. These are both very different units.

you need to activate the developing mode (Entlicklermodus) on the MIB, it can be done with VCDS or OBDeleven. After that you need to hold the MENU Button a couple of seconds and you are in the Service Mode. Now after enabling the developing mode you need to see there a function called "Test mode" and there you have this Trace Functions

Yes the Delphi Units are different. I also heard that they are not so good for retrofitting and unlocking.

the MIB2 HIGH Units are also from Harman. So maybe they have the same Quadlock Pinout like the Porsche Units.

**Comment by jannikuhl** [comment]
Found it, thanks. I was always looking in the green menu.

Porsche PCM and MIB2 HIGH are nearly the same. Both from Harman and can be patched the same way. So I think the pinout is also the same.

I'm currently not up-to-date: Is it possible to upload custom green menus already?
Anyone tried it the same way it works on MIB2 high?

**Comment by Vavulinalex** [comment]
Hello. I want to study the mib2std Technisat file system. I tried connecting via uart. Unsuccessfully. Technisat does not have a sequential shell. I want to try using telnet. Can you tell me what BDM is? I want to activate telnet.

**Comment by jannikuhl** [comment]
You're right, Technisat does not have a serial shell. What you need to do is to read the EMMC chip, activate telnet and flash the whole system back to the chip. As described in this this guide: https://forum.xda-developers.com/general/connected-car/success-to-hack-technisat-mib2-t3584185

BDM can be used instead of soldering. You need BDM probes to connect directly to the circuit board. Here some pictures:

https://contestimg.wish.com/api/webimage/5dc6806fe362821086a79e51-0-large?cache_buster=66a2ba98886f0bf85989036c6d6fd5c8

https://www.dhresource.com/0x0s/f2-albu-g9-M00-38-BB-rBVaWFwHnM-AICmuAAFDktohDAA328.jpg

**Comment by Vavulinalex** [comment]
Thanks. If necessary, I can share the instructions for mib2 High. There are root passwords for different firmware and instructions on how to work with fec/swap

**Comment by yox2019** [comment]
How did yoy active telnet ? ... inetd ?

**Comment by yox2019** [comment]
... /etc/system/enum/devices/net ;)

device(usb, ven=2001,dev=3c05) # D-Link DUB-E100 big version
device(usb, ven=2001,dev=1a02) # D-Link DUB-E100 small version
device(usb, ven=0b95,dev=772b) # Edimax EU-4208 small version
device(usb, ven=0b95,dev=7720) # Edimax EU-4207 big version
device(usb, ven=0b95,dev=1780) # DELOCK 61969 USB 2.0 Gigabit LAN Adapter
   waitfor(/dev/socket)
   driver (mount -Tio-pkt -o "verbose,phy_check,busnum=$(busno),devnum=$(devno)" devnp-asix.so)
   start/wait(if_up -p ax0)
   start(ifconfig ax0 192.168.1.4)
   requires(inetd,)
   requires(qconn,)

device(usb)
   echo("No match found for device ven=$(ven), dev=$(dev), class=$(class), busno=$(busno), devno=$(devno), cfg=$(cfg), iface=$(iface), msven=$(msven), mscomp=$(mscomp), mssubcomp=$(mssubcomp)" )

**Comment by flipidus** [comment]
> device(usb, ven=2001,dev=3c05) # D-Link DUB-E100 big version
> device(usb, ven=2001,dev=1a02) # D-Link DUB-E100 small version
> device(usb, ven=0b95,dev=772b) # Edimax EU-4208 small version
> device(usb, ven=0b95,dev=7720) # Edimax EU-4207 big version
> device(usb, ven=0b95,dev=1780) # DELOCK 61969 USB 2.0 Gigabit LAN Adapter

Are these the working USB to LAN Interface Adapter for MIB2 STD/HIGH?

**Comment by yox2019** [comment]
> > device(usb, ven=2001,dev=3c05) # D-Link DUB-E100 big version
> > device(usb, ven=2001,dev=1a02) # D-Link DUB-E100 small version
> > device(usb, ven=0b95,dev=772b) # Edimax EU-4208 small version
> > device(usb, ven=0b95,dev=7720) # Edimax EU-4207 big version
> > device(usb, ven=0b95,dev=1780) # DELOCK 61969 USB 2.0 Gigabit LAN Adapter
> 
> Are these the working USB to LAN Interface Adapter for MIB2 STD/HIGH?

... its from PQ unit you have to try it on MIB2 STD/HIGH I have any to test  ;)
... have you telnet connection with "login" promt ?

**Comment by flipidus** [comment]
Okay, then i will look forward to buy one of these USB to LAN Adapters and then i can check if i can get a telnet promt. are there some special subnet and ip adress static settings i need to set for this network adapter? i dont think the MIB2 will host a DHCP

**Comment by jannikuhl** [comment]
You need to enable it in the green menu. Then you can connect with D-Link. You can read off the required network settings in the green menu.

**Comment by yox2019** [comment]
> You need to enable it in the green menu. Then you can connect with D-Link. You can read off the required network settings in the green menu.

I am able to connect via D-Link  (192.168.1.4) but only on few ports then I can see logs but connection on port 23 is refused, there is another network 10.X.x.x did you get "login" promt on 192.... or 10... ?

**Comment by flipidus** [comment]
Okay, thanks for the information, i will look in the GEM for that IP Settings and i ordered a D-Link DUB-E100 USB to LAN adapter, the smaller black version.

**Comment by jannikuhl** [comment]
You need to connect to port 23 and need the following adapter settings:

- IP: 192.168.1.100
- Subnet: 255.255.255.0

By the way, does anyone have experience recovering Delphi units? I bricked mine today with a invalid SWaP File.

**Comment by yox2019** [comment]
> You need to connect to port 23 and need the following adapter settings:
> 
> * IP: 192.168.1.100
> * Subnet: 255.255.255.0

... sorry for the question are the D-Link settings or the ethernet card in the computer?

**Comment by yox2019** [comment]
> By the way, does anyone have experience recovering Delphi units? I bricked mine today with a invalid SWaP File.

... you have to find the way tu put this unit in "emrgency mode" then reflash it with software already installed

**Comment by jannikuhl** [comment]
On the ethernet card of the PC.

I don't think it will go into emergency mode as it is in a constant boot loop.

**Comment by yox2019** [comment]
> On the ethernet card of the PC.

Thanks, and you login on address displayed in green menu e.g in my unit 192.168.1.4  ?
 
> I don't think it will go into emergency mode as it is in a constant boot loop.

it doesn't matter, you can always turn on emergency mode, even if the unit is working properly, you just need to know how...

**Comment by jannikuhl** [comment]
Yes, exactly. Login is root and there is no password, just press enter.

@yox2019 Do you know how to enter emergency mode? It seems to be the last chance for this unit before it needs soldering.

**Comment by yox2019** [comment]
> Yes, exactly. Login is root and there is no password, just press enter.
... THX I will try but I afraid in PQ unit it won't working

> @yox2019 Do you know how to enter emergency mode? It seems to be the last chance for this unit before it needs soldering.

... no, unfortunately but I think it have to be similar as Technisat any way you need terminal connection usb/uart and putty to be able put this unit in emergency mode

**Comment by jannikuhl** [comment]
Do you talk about Technisat or Delphi?
uart does only work on Delphi and Harman units. Technisat does not have any serial port open, you need to read the emmc, e.g. using BDM. There is currently no other option. PQ is Technisat.

**Comment by flipidus** [comment]
you have ZR devices from Techniat (Preh) / Delphi / Harman and PQ devices from Technisat. I think the ZR devices from Technisat are to handle the same as the PQ devices. Im waiting for this USB to LAN Adapter from ebay, so i can also test the Telnet function on my Preh device

i heard the Delphi Devices are not so hacking-friendly but i cannot proove if this is true.

i have a productive Technisat/Preh MIB2 in my Car and a test-device from technisat (without Navigation) for testing purposes. but for my test desk i still need som wiring stuff to connect the MIB2 with the ABT (single wires, HSD cable, plugs, etc...) 

does anyone know how the component protection is going ON, when you use a MIB2 without CAN communication? Running time? Boot counter?

**Comment by yox2019** [comment]
> Yes, exactly. Login is root and there is no password, just press enter.


it doesn't work like I thought and that's why:
start(ifconfig ax0 192.168.1.4)
**requires**(inetd,)
**requires**(qconn,)

I'm talking about Technisat PQ unit ;)

=~=~=~=~=~=~=~=~=~=~=~= PuTTY log 2020.05.09 13:07:04 =~=~=~=~=~=~=~=~=~=~=~=


[41m-Welcome to TechniBoot-[0m
SVN Rev: 490068 - Jun 18 2015 / 16:46:28
Variant: PQ stdNav 2GB LPDDR3
CAAM: CAAM_CSTA  0x00000202 - trusted Mode
CAAM: init -> ok
CAAM: open already instantiation ring 0!
CAAM: open ok
Image verified
CAAM: clock 0x00018D00
type=00000004, cksum=00009803, address=10800000, length=002EE9CC
type=00000006, cksum=00000006, address=10800000, length=00000000


iMX6.QNX.LoadImage.ready: 0x13107EAD 
Enabling only 2 CPUs
L2 cache enabled
CPU0: L1 Icache: 1024x32
CPU0: L1 Dcache: 1024x32 WB
CPU0: VFP-d32 FPSID=41033094
CPU0: NEON MVFR0=10110222 MVFR1=01111111
CPU0: 412fc09a: Cortex A9 MPCore rev 10 996MHz
Board version: PQ/6 v12

Detected i.MX6 Dual/Quad, revision TO1.5

**Comment by flipidus** [comment]
> device(usb, ven=2001,dev=1a02) # D-Link DUB-E100 small version
> device(usb, ven=0b95,dev=1780) # DELOCK 61969 USB 2.0 Gigabit LAN Adapter

i got this two adapters and try to get a LAN Connection between my notebook and the MIB2 STD ZR (VW, Preh, H41, SW0478)

i was in the debug settings in the GEM and there are IP adresses listed but i was not able to get a connection, the last IP adress in the list was not readable completely and only appeard when the USB to LAN Adapter was connected and the MIB2 was not pingable.

Maybe somebody have some hints?

Thanks

**Comment by jannikuhl** [comment]
You cannot connect to a Technisat (ZR) MIB2 with these adapters. You need to read eMMc and enable Telnet. As I have described above.

**Comment by Vavulinalex** [comment]
Do you have an emmc Pinout ?
![IMG_20200629_213102](https://user-images.githubusercontent.com/66567681/86048478-43c41100-ba59-11ea-9609-fc0fbf67d283.jpg)
![IMG_20200629_212326](https://user-images.githubusercontent.com/66567681/86048481-458dd480-ba59-11ea-8944-3c019b807399.jpg)

**Comment by yox2019** [comment]
> You cannot connect to a Technisat (ZR) MIB2 with these adapters. You need to read eMMc and enable Telnet. As I have described above.

what do you mean "enable telnet"? I am able to read emmc but as I asked before how do you think "enable telnet"?

**Comment by Vavulinalex** [comment]
> > Вы не можете подключиться к Technisat (ZR) MIB2 с этими адаптерами. Вам нужно прочитать eMMc и включить Telnet. Как я уже описал выше.
> 
> что значит "включить telnet"? Я могу читать emmc, но как я уже спрашивал, как вы думаете, «включить telnet»?

You don't understand anything. You have already been told several times. To use telnet, you must first activate the shell.

https://forum.xda-developers.com/general/connected-car/success-to-hack-technisat-mib2-t3584185 Read the first post

**Comment by flipidus** [comment]
> 
> 
> > > Вы не можете подключиться к Technisat (ZR) MIB2 с этими адаптерами. Вам нужно прочитать eMMc и включить Telnet. Как я уже описал выше.
> > 
> > 
> > что значит "включить telnet"? Я могу читать emmc, но как я уже спрашивал, как вы думаете, «включить telnet»?
> 
> You don't understand anything. You have already been told several times. To use telnet, you must first activate the shell.
> 
> https://forum.xda-developers.com/general/connected-car/success-to-hack-technisat-mib2-t3584185 Read the first post

In this XDA Forum Post, it is described how to activate a SERIAL shell. Serial and Telnet is not the same.

**Comment by Vavulinalex** [comment]
> > > > Вы не можете подключиться к Technisat (ZR) MIB2 с этими адаптерами. Вам нужно прочитать eMMc и включить Telnet. Как я уже описал выше.
> > > 
> > > 
> > > что значит "включить telnet"? Я могу читать emmc, как вы думаете, «включить telnet»?
> > 
> > 
> > Вы ничего не понимаете. Вам уже сказали несколько раз. Чтобы использовать telnet, вы должны сначала активировать оболочку.
> > https://forum.xda-developers.com/general/connected-car/success-to-hack-technisat-mib2-t3584185 Прочитайте первый пост
> 
> В этом сообщении на форуме XDA описано, как активировать SERIAL shell. Serial и Telnet - это не одно и то же.

Until you have either a serial connection or a telnet connection.

**Comment by yox2019** [comment]
> > > Вы не можете подключиться к Technisat (ZR) MIB2 с этими адаптерами. Вам нужно прочитать eMMc и включить Telnet. Как я уже описал выше.
> > 
> > 
> > что значит "включить telnet"? Я могу читать emmc, но как я уже спрашивал, как вы думаете, «включить telnet»?
> 
> You don't understand anything. You have already been told several times. To use telnet, you must first activate the shell.
> 
> https://forum.xda-developers.com/general/connected-car/success-to-hack-technisat-mib2-t3584185 Read the first post


you are wrong, I understand and repeat it again, even if you get a shell, you will not be able to login via telnet, at least in the Technisat PQ Unit

**Comment by Vavulinalex** [comment]
> > > > Вы не можете подключиться к Technisat (ZR) MIB2 с этими адаптерами. Вам нужно прочитать eMMc и включить Telnet. Как я уже описал выше.
> > > 
> > > 
> > > что значит "включить telnet"? Я могу читать emmc, как вы думаете, «включить telnet»?
> > 
> > 
> > Вы ничего не понимаете. Вам уже сказали несколько раз. Чтобы использовать telnet, вы должны сначала активировать оболочку.
> > https://forum.xda-developers.com/general/connected-car/success-to-hack-technisat-mib2-t3584185 Прочитайте первый пост
> 
> вы не правы, я понимаю и повторяю это снова, даже если вы получите оболочку, вы не сможете войти через telnet, по крайней мере, в Technisat PQ Unit

I won't argue with you. try again, but you are not going in the right direction.
Telnet will work. This is evidenced by the facts.
1. jannikuhl wrote about this at the very beginning.
2. I communicated with a hacker. He said it was free. This solution works.
And offered a paid solution (500 euro). Software package

**Comment by yox2019** [comment]
> > > > > Вы не можете подключиться к Technisat (ZR) MIB2 с этими адаптерами. Вам нужно прочитать eMMc и включить Telnet. Как я уже описал выше.
> > > > 
> > > > 
> > > > что значит "включить telnet"? Я могу читать emmc, как вы думаете, «включить telnet»?
> > > 
> > > 
> > > Вы ничего не понимаете. Вам уже сказали несколько раз. Чтобы использовать telnet, вы должны сначала активировать оболочку.
> > > https://forum.xda-developers.com/general/connected-car/success-to-hack-technisat-mib2-t3584185 Прочитайте первый пост
> > 
> > 
> > вы не правы, я понимаю и повторяю это снова, даже если вы получите оболочку, вы не сможете войти через telnet, по крайней мере, в Technisat PQ Unit
> 
> I won't argue with you. try again, but you are not going in the right direction.
> Telnet will work. This is evidenced by the facts.
> 
> 1. jannikuhl wrote about this at the very beginning.
> 2. I communicated with a hacker. He said it was free. This solution works.
>    And offered a paid solution (500 euro). Software package


... oo yes;))) How about I have a unit with an unlocked console and telnet does not work? ;) Do you want "shadow" file and "passwd"?

Here you are:
passwd:
"Root: x: 0: 0: Superuser: /: / bin / sh
ftp :: 0: 0: FTP User: /: / bin / sh
mdnsd: x: 16: 7: mdnsd: / nonexistent: / bin / false "

shadow:
"root: UGCfzJmLNga36:98:0:0"

... and now give me the telnet port where I can use my login and password ;)))

I repeat again, if you do not edit the firewall properly, you will not log in via telnet

... aha "pf.conf" if you know what is it ;)

ext_wlan_if=wlan0
ext_ppp_if=ppp0

set block-policy drop
set skip on lo0
set reassemble yes
scrub in all

nat on $ext_ppp_if from !($ext_ppp_if) to any -> ($ext_ppp_if)

block in from 127.0.0.0/8 to any

#block multicasts IPv4 (224.0.0.0 to 239.255.255.255) except port 1900
block in proto { tcp, udp } from any to 224.0.0.0/4 port !=1900

block in all
pass in quick on tsd_imx60 all
pass in quick on tsd_j50 all
pass in quick on lo all
pass out quick all keep state

#HMI Logging
pass in quick on ax proto tcp from any to any port 15001 keep stat

anchor port_rules
load anchor port_rules from "/tsd/var/networking/port_rules.conf"

... do you want telnet you can only log in to port 15361 and see "flying logs";)))

pf_testmode.conf: 

set block-policy drop
set skip on lo0
#set fingerprints "/etc/pf.os"

#block in all
pass in quick proto icmp from any to any
pass in quick on ncm all
pass in quick on tsd_imx60 all
pass in quick on tsd_j50 all
pass in quick on lo all
pass out quick all keep state

pass in proto { udp , tcp } from any to any port 15361 keep state

#Testschnittstelle
pass in quick proto { tcp, udp } from any to any port 1234 keep state

... maybe there will be someone who can correctly set up a firewall? ;)))

**Comment by jannikuhl** [comment]
Does anyone know how to read eMMC or Delphi Unit or even better how to push it into emergency mode?
My Unit does no longer allow Telnet and reboots several times.

**Comment by cfectus** [comment]
> Thanks. If necessary, I can share the instructions for mib2 High. There are root passwords for different firmware and instructions on how to work with fec/swap

Do you have this info for Delphi devices also? I need it to give it a try...

**Comment by Vavulinalex** [comment]
> > Thanks. If necessary, I can share the instructions for mib2 High. There are root passwords for different firmware and instructions on how to work with fec/swap
> 
> Do you have this info for Delphi devices also? I need it to give it a try...
look there:
␡

**Comment by shadowswan** [comment]
> > Thanks. If necessary, I can share the instructions for mib2 High. There are root passwords for different firmware and instructions on how to work with fec/swap
> 
> Do you have this info for Delphi devices also? I need it to give it a try...
> look there:
> https://mega.nz/folder/DRZQ1a6b#Y2BJ_EBCDGCj2PHzHfTs2Q

do you have a version of this for tstat?

**Comment by Vavulinalex** [comment]
Technisat? Of course
❌

**Comment by Vavulinalex** [comment]
Who deletes my links? Do you feel sorry for that?
Guys write in PM

**Comment by mattcabb** [comment]
I'm deleting them. I want this repo to stay clean.

**Comment by ianidor** [comment]
Hi 

Loving this informative thread 
I’m currently trying to also hack car play and unlock CP in these units 
As I understand the emmc dumping is only to read the emmc to run QNX virtually to then edit code to include a telnet console......
It seams easy and there’s is pinouts on the Porsche/xda thread above (via mega upload)
And for the bricked unit it also has pinouts for shorting pinouts in the units and driving into emergency mode 
Mattcabb is it okay to share the megaupload link?

**Comment by ianidor** [comment]
Delete if not allowed 

Harman (edited)

✂️

Technisat
✂️
Delphi
✂️

**Comment by flipidus** [comment]
> Delete if not allowed
> 
✂️

Harman Link is not working, asks for a encryption key

**Comment by shadowswan** [comment]
They have already been deleted previously by him.
It’s fairly straight forward to unlock CP now and I assume that a lot of the techniques in these exploits could be used to run this toolbox. It’s well documented now on how to dump the files from the device on both technistat and Delphi.

**Comment by ianidor** [comment]
Yes I know but there’s pinouts for dumping emmc without removing

**Comment by ianidor** [comment]
> > Delete if not allowed
> > ✂️
> 
> Harman Link is not working, asks for a encryption  
✂️

**Comment by EvoSems** [comment]
Can anyone explain easily how to pinout reset Delphi into fresh emergency mode. I am stuck in a emergency loop but it always asks for the specific firmware it failed on when loading fw at start. And of course it accept the correct firmware it failed on. Loads 4 more files then says complete but then opens back up into emergency n the same loop. Different firmware’s give error that it’s not the firmware that failed.

**Comment by dnoermann** [comment]
Hallo to all, first, thanks for sharing all this.

I have Technisat MIB2STD FW 0245
I Enabled telnet, port 23 access in the firewall with adding 
"pass in quick proto tcp from any to any port 23 keep state" in tsd/etc/networking/pf.conf

Enabled Console in startup, like @jannikuhl did explained it.

nmap now shows telnet as not blocked by the firewall, but still closed, so the telnetd is not running ....
any hints, how to enable telnetd ?

**Comment by jannikuhl** [comment]
@dnoermann Can you enter commands using serial connection? I use a serial->USB adapter with a FTDI Chip.

**Comment by dnoermann** [comment]
@jannikuhl from what i read in this comments, my understandend was that the serial console on the back connector is not available on the PQ Techisat model. I their for did not try this till now, will try it.
Update: Serial Console Works :)
I needed to add in startup:
echo /net/J5/dev/ser1 \"/bin/login -f root\" qansi-m on > /tmp/ttys
/sbin/tinit -f /tmp/ttys &

**Comment by ianidor** [comment]
Excellent work 
It appears all this work is kept quiet to the common DIY guy 
I’m wondering if anyone had worked on the PQ I’m currently in emergency mode and can’t get it to take any firmware?? 
Anyone done this ??

**Comment by dnoermann** [comment]
To start telnetd, after enabling Telnet in Firewall:
/usr/sbin/telnetd -debug 23 &
root pwd for 0245 FW PQ Technisat is "root"

**Comment by ianidor** [comment]
Thanks dnoermann any idea of a command to make it take an update from the sd card ?

**Comment by fikaa73** [comment]
Hi guys.
I have TechniSat unit(from my research)
MST2_EU_VW_ZR_P0241T
3Q0035846

Can we join knowledge about that unit here?
From my testing, by connecting it to wifi, unit is pingable, and has open 5 ports(23 nor 123 is open)
If you try to telnet to any of these ports, you just get flying logs and that’s it, no network connection can be made.
So, I guess only option is to enable telnet through serial. How is that done, and what hardware is needed?
@ianidor maybe could help on this one :)

**Comment by jannikuhl** [comment]
@fikaa73 Serial will also not work, you will only be able to get output but serial input is disabled.
You have to read eMMC or if it is a Nav unit inject the configuration change as a firmware update ;)

**Comment by fikaa73** [comment]
> @fikaa73 Serial will also not work, you will only be able to get output but serial input is disabled.
> You have to read eMMC or if it is a Nav unit inject the configuration change as a firmware update ;)

Yeah, it’s unit with nav, it can be updated from usb, sd card and cd.
Can someone please guide me through this, as I really wouldn’t like to brick it😳
Which files should I have to update it anyways?

**Comment by ianidor** [comment]
fikaa73 
Don’t worry if you brick it 
I can recover 👌

**Comment by fikaa73** [comment]
Just don’t ask for money please, I already met people who do that for money 😂
I’d be glad if someone could guide me shortly through this, which files and what to edit? :)

**Comment by Vavulinalex** [comment]
> Только денег не проси, я уже встречал людей, которые делают это за деньги 😂
> Я был бы рад, если бы кто-нибудь вкратце объяснил мне, какие файлы и что редактировать? :)

Hi. I wrote here the links to all the instructions. But they are removed from here. People make money on this and don't want to distribute instructions

**Comment by fikaa73** [comment]
> > Только денег не проси, я уже встречал людей, которые делают это за деньги 😂
> > Я был бы рад, если бы кто-нибудь вкратце объяснил мне, какие файлы и что редактировать? :)
> 
> Hi. I wrote here the links to all the instructions. But they are removed from here. People make money on this and don't want to distribute instructions

Hello friend. Would you mind emailing me instructions for my unit to fvukotic@gmail.com?
I have guy in my city asking for 150€ to unlock my unit. I really wouldn’t throw money on that, when I know I can do it myself.

**Comment by mattcabb** [comment]
> > Только денег не проси, я уже встречал людей, которые делают это за деньги 😂
> 
> > Я был бы рад, если бы кто-нибудь вкратце объяснил мне, какие файлы и что редактировать? :)
> 
> 
> 
> Hi. I wrote here the links to all the instructions. But they are removed from here. People make money on this and don't want to distribute instructions

I'm removing those links. I don't make money on this. In fact everything that I know is shared here, in this repo, for free.
There are legal reasons why I don't want those links in this repo. It contains copyright protected software as well as tools to bypass Component Protection = activating stolen devices.
You can share what you want but not here.

**Comment by flipidus** [comment]
> > Только денег не проси, я уже встречал людей, которые делают это за деньги 😂
> > Я был бы рад, если бы кто-нибудь вкратце объяснил мне, какие файлы и что редактировать? :)
> 
> Hi. I wrote here the links to all the instructions. But they are removed from here. People make money on this and don't want to distribute instructions

You can do your own thread in github, if you want to share these files. but make the instructions a little bit less confusing.

**Comment by cfectus** [comment]
> > Только денег не проси, я уже встречал людей, которые делают это за деньги 😂
> > Я был бы рад, если бы кто-нибудь вкратце объяснил мне, какие файлы и что редактировать? :)
> 
> Hi. I wrote here the links to all the instructions. But they are removed from here. People make money on this and don't want to distribute instructions

Thanks once more for files :) But i could not make the work with them cause it seems something is missing in them. How to edit the SWaP files for versions, and how to make fec files for features. if you know how to do, can you share it please by cemkokten@hotmail.com. I have two units to do the retrofit on my car and for a friend, one is preh and the other one is delphi, and as i see preh units need some hardware work do do the work, right? They really make money on it and they request a lot for it :(

**Comment by fikaa73** [comment]
> > > Только денег не проси, я уже встречал людей, которые делают это за деньги 😂
> > > Я был бы рад, если бы кто-нибудь вкратце объяснил мне, какие файлы и что редактировать? :)
> > 
> > 
> > Hi. I wrote here the links to all the instructions. But they are removed from here. People make money on this and don't want to distribute instructions
> 
> Thanks once more for files :) But i could not make the work with them cause it seems something is missing in them. How to edit the SWaP files for versions, and how to make fec files for features. if you know how to do, can you share it please by [cemkokten@hotmail.com](mailto:cemkokten@hotmail.com). I have two units to do the retrofit on my car and for a friend, one is preh and the other one is delphi, and as i see preh units need some hardware work do do the work, right? They really make money on it and they request a lot for it :(

If you find out please email me on fvukotic@gmail.com I’m looking for same thing.
Did you succeed in updating your mib2?

**Comment by cfectus** [comment]
> 
> If you find out please email me on [fvukotic@gmail.com](mailto:fvukotic@gmail.com) I’m looking for same thing.
> Did you succeed in updating your mib2?

Sure i will share, not tried to update yet, i could't make my mind clear on instructions cause they seem to be not complete.

**Comment by yox2019** [comment]
> > > Только денег не проси, я уже встречал людей, которые делают это за деньги 😂
> > > Я был бы рад, если бы кто-нибудь вкратце объяснил мне, какие файлы и что редактировать? :)
> > 
> > 
> > Hi. I wrote here the links to all the instructions. But they are removed from here. People make money on this and don't want to distribute instructions
> 
> ... They really make money on it and they request a lot for it :(


... so use your brain, create a solution and share for free what's the problem ?

**Comment by cfectus** [comment]
> ... so use your brain, create a solution and share for free what's the problem ?

:) There is no problem. I respect the guy who found the solution, and also respect what he requests. But i'm only talking about the ones got the solution for free but request money for it (if the founder not allowed him/her to do) not any other :) And thanks to everbody learn the solution for free and share it.
As i did many times before, if i found more info i will definitely share it. Sorry if i am misunderstood.

**Comment by fikaa73** [comment]
I succeeded with patching swap, i have generated swap codes, but here's the problem
I don't have vcp
Could anyone assist me on how to push swap codes through update?

**Comment by ianidor** [comment]
Hi 
Just bought vcp as I’m at the same point 
Are you uk based ?
I’m happy to help

**Comment by fikaa73** [comment]
> Hi
> Just bought vcp as I’m at the same point
> Are you uk based ?
> I’m happy to help

I’m from Montenegro actually :)
I don’t have vcp, but I know there’s a way of pushing swap codes through update, so u don’t need vcp.
I can send you swap generator and instructions for vcp

**Comment by ianidor** [comment]
Yes there is but not sure you can do it in these Technisat units 
Do let me know if you find a way 
Thanks
 ian

**Comment by fikaa73** [comment]
> Yes there is but not sure you can do it in these Technisat units
> Do let me know if you find a way
> Thanks
> ian

Well I guess I’m stuck with vcp now.
But wait, if I can alter files, is there any other modification I can do, besides of swap?

EDIT:Is anyone able to explain how to activate Telnet on Technisat? I added line to startup and pushed it through update, but nothing happened, telnet still closed.

**Comment by dnoermann** [comment]
> EDIT:Is anyone able to explain how to activate Telnet on Technisat? I added line to startup and pushed it through update, but nothing happened, telnet still closed.

You need to open Firewall for Port 23
Add following line to pf.conf from tsd/etc/networking/
pass in quick proto tcp from any to any port 23 keep state

**Comment by shadowswan** [comment]
Is the only reason you want to do that for swap? If you have patched the swap file you can load them up using OBDEleven much easier

**Comment by mattcabb** [comment]
...you can also insert `signed_exception_list.txt` file with FSC codes that you wish to enable.

**Comment by fikaa73** [comment]
@shadowswan yeah, thought only way is either that or vcp. How can i do it with obd11? 😦
@mattcabb oh, awesome. But I still need to insert that files somehow. Can i edit pf.conf and hash it with hash generator and push regularly through update?
My pf.conf is in `cpu\networkmgr_sec\12\default\cpu_networkmgr_sec_12_default.tar.gz\cpu_networkmgr_sec_12_default.tar\tsd\etc\networking\`
is that path okay, and I can push it through update just as I did when I was patching swap file?
And can you give me scheme of `signed_exception_list.txt` please? Do I just add new code, or I must also have old codes?

**Comment by shadowswan** [comment]
@fikaa73 ensure it is in EOL mode, Security access 20103. 
Go to 'Transmit code unlock function SWaP' and enter FEC code
Execute 'unlock function SWaP'
Reboot the headunit 
'Eh voilà'

Some of my translations might be a bit off but you will be able to figure them out.

**Comment by fikaa73** [comment]
> @fikaa73 ensure it is in EOL mode, Security access 20103.
> Go to 'Transmit code unlock function SWaP' and enter FEC code
> Execute 'unlock function SWaP'
> Reboot the headunit
> 'Eh voilà'
> 
> Some of my translations might be a bit off but you will be able to figure them out.

Oh, I didn’t know that’s possible.
What’s EOL mode and how do I come to it?:) 
Yeah I’d say I’ll figure it out

**Comment by shadowswan** [comment]
Select 5f then change service then EOL

**Comment by fikaa73** [comment]
> Select 5f then change service then EOL

Owh, thank you, didn't know that.
Awesome thing, I'll try as soon as OBDEleven arrives.

EDIT:How do I get my `VCRN` if I don't have it in engineering menu?

**Comment by shadowswan** [comment]
Here is not really the place for all these questions, VCRN can be found in the enabled codes menu by selecting one of the codes. 
Have you got the mega link? You can find it easily online on mhh or CT forums and it answers all of this.

**Comment by fikaa73** [comment]
> Here is not really the place for all these questions, VCRN can be found in the enabled codes menu by selecting one of the codes.
> Have you got the mega link? You can find it easily online on mhh or CT forums and it answers all of this.

- [x] I don't have third option in menu where FEC codes should be, only update and hmi tails. And yeah, I got mega links, updated my unit, patched swap.
- [x] I ask so many questions because I really don't want to mess up anything, and also that info will 100% be useful for someone else. 
- [x] I'm stuck with signed_exception_list.txt and what's content of that file
- [x] @mattcabb can you give me usable example of signed_exception_list.txt please?

Edit:Passing signed_exception_list.txt with just signatures and no FEC codes activates all codes available. Problem solved, CarPlay activated

**Comment by johnxter** [comment]
@fikaa73 can you help me with signed_exception_list.txt please ?
I'm stuck with my skoda, I had SK ZR 0245T and after update to 0253 and patch, all my fecs gone, no BT, no nav now
how can I activate back only with sd card ?
thank you

**Comment by fikaa73** [comment]
> @fikaa73 can you help me with signed_exception_list.txt please ?
> I'm stuck with my skoda, I had SK ZR 0245T and after update to 0253 and patch, all my fecs gone, no BT, no nav now
> how can I activate back only with sd card ?
> thank you

@johnxter email me on fvukotic@gmail.com please, no links are allowed here

**Comment by fikaa73** [comment]
> SWaP

Hi @shadowswan
Can you explain me how to get car to EOL mode through vcds?
There is adaptation in 5f but its value is 00 00 00 and new value must be in that format.
What should I do?

**Comment by shadowswan** [comment]
Needs to be vcp / obd11 / odis. VCDS doesn’t support EOL afaik

**Comment by fikaa73** [comment]
> Needs to be vcp / obd11 / odis. VCDS doesn’t support EOL afaik

Thank you.
Do I insert whole generated swap code, or only 8 digit fec?

**Comment by shadowswan** [comment]
Yes enter whole code then execute the second command

**Comment by fikaa73** [comment]
> Yes enter whole code then execute the second command

Awesome.
Can I get your email please?

**Comment by EvoSems** [comment]
Having a problem with security on a polo gti 6C. With VCDS security code is S12345

I tried S12345 as well as 20103
But get error Key Refused RC 35 Key Incorrect. 

I’ve tried this on 2 MIB units on the same car. 

I am a little new to VCP. my process is as follows. (Let me know if this is correct.)

Load Program.
Select More. Manual. 5F Start
Select SE26X (defaults to my Car I guess?)
Select EOL VW
SEC.Access. 
Enter Login either S12345/20103
Get error.

**Comment by fikaa73** [comment]
> Having a problem with security on a polo gti 6C. With VCDS security code is S12345
> 
> I tried S12345 as well as 20103
> But get error Key Refused RC 35 Key Incorrect.
> 
> I’ve tried this on 2 MIB units on the same car.
> 
> I am a little new to VCP. my process is as follows. (Let me know if this is correct.)
> 
> Load Program.
> Select More. Manual. 5F Start
> Select SE26X (defaults to my Car I guess?)
> Select EOL VW
> SEC.Access.
> Enter Login either S12345/20103
> Get error.

Did you try first security access then eol?

**Comment by mattcabb** [comment]
Security Access S12345 is just for VCDS. It doesn't send the code to the 5F unit. Instead it changes VCDS diagnostic session to <strike>EOL</strike> developer mode.
If you are using VCDS then use SA S12345. If you have VCP or OBDeleven then just change diagnostic session.

**Comment by EvoSems** [comment]
Oh I am using VCP. 
I will try security access first tonight. But I have tried this a few times now. And it seems it doesn’t accept the key. Then I am locked out for maybe 30 mins before trying again. 

Maybe I am doing the process wrong. I can’t seem to find any videos on using the vcp interface yet. So I’m a bit trial and error each time. 

I’m assuming once the code is entered correct I can open EOL. If I select it now. Nothing happens. 

Thanks for the guidance chaps.

**Comment by SqugSCX** [comment]
Not sure if this is the right place to ask, but how do you actually get this solution working? As in reading EMMC using the SD-card method on a Discover Media unit? Or can I just update the unit with a different firmware (0247T vs 0253T) that is already patched?

**Comment by fikaa73** [comment]
@shadowswan check this out
![C7CC1301-7F1C-4EFF-9925-D9935C62DE92](https://user-images.githubusercontent.com/72288204/102725601-32790480-4318-11eb-9a8d-5b8901c869f6.jpeg)

**Comment by shadowswan** [comment]
That’s the wrong place, will look tomorrow but it’s something like transmit code, you put the code in and then You run the release which is the one you are in there

EDIT:

its under adaptation and is ‘transfer of release code for a swap function’ paste code in there and then go back to the one you had and change it to active.

**Comment by mattcabb** [comment]
@shadowswan so the procedure for OBDeleven is:
5F > Change service > EOL
5F > Adaptation > Transfer of release code for a SWaP Function > paste_the_code > save
5F > Basic Settings > Release of SWaP function > paste_the_code > set status to active
Reboot, long press MENU, check if code is valid.

Is that correct?

**Comment by shadowswan** [comment]
Almost, for the release of swap function you just change it to active and save that. 

Obviously the swap file and the swdownload need to have been patched first for it to accept it

**Comment by mattcabb** [comment]
@shadowswan system.swap needs to be patched to ignore the signature.
Why system.swdownload needs to be patched as well?

**Comment by shadowswan** [comment]
Sorry you are correct I was thinking of software update too. 

@fikaa73 is it working now?

**Comment by fikaa73** [comment]
@shadowswan I don't have hands on my car now, but will look for it tomorrow. That one makes sense, I'm looking just now for VCP instructions and that's correct sequence of steps. Hope it will work :)
SWaP needs to be patched because this is forced way to push SWaP codes, just like you would do with VCP. If you take a look, SWaP from generator is same for every car, it just differs for VCRN and VIN, but I think some other part should be different, like signature or something. 
In basic settings, "Release swap function" has placeholder 00-FF, you sure I just run it without any value?

@mattcabb thank you for looking into this too, and explaining how patches work :)
Thank you guys very much, I will inform you what have I done.

**Comment by shadowswan** [comment]
> @shadowswan I don't have hands on my car now, but will look for it tomorrow. That one makes sense, I'm looking just now for VCP instructions and that's correct sequence of steps. Hope it will work :)
> SWaP needs to be patched because this is forced way to push SWaP codes, just like you would do with VCP. If you take a look, SWaP from generator is same for every car, it just differs for VCRN and VIN, but I think some other part should be different, like signature or something.
> In basic settings, "Release swap function" has placeholder 00-FF, you sure I just run it without any value?

VCP instructions are in the mega folder which I think you had? 

It’s been a while since I’ve done the OBD11 but 99% that you just leave it place and choose active.

**Comment by fikaa73** [comment]
> > @shadowswan I don't have hands on my car now, but will look for it tomorrow. That one makes sense, I'm looking just now for VCP instructions and that's correct sequence of steps. Hope it will work :)
> > SWaP needs to be patched because this is forced way to push SWaP codes, just like you would do with VCP. If you take a look, SWaP from generator is same for every car, it just differs for VCRN and VIN, but I think some other part should be different, like signature or something.
> > In basic settings, "Release swap function" has placeholder 00-FF, you sure I just run it without any value?
> 
> VCP instructions are in the mega folder which I think you had?
> 
> It’s been a while since I’ve done the OBD11 but 99% that you just leave it place and choose active.

Yep, was in front of me whole time, just like when I was looking for signed_exception_list.txt for 10 days and accidentally discovered I had it in CP_OFF folder whole time 😂 I'm really surprising myself sometimes..
Can't wait to get in the car to try it

**Comment by shadowswan** [comment]
@fikaa73 enjoy!

**Comment by EvoSems** [comment]
> Security Access S12345 is just for VCDS. It doesn't send the code to the 5F unit. Instead it changes VCDS diagnostic session to EOL developer mode.
> If you are using VCDS then use SA S12345. If you have VCP or OBDeleven then just change diagnostic session.

Guys I'm having such a headache with VCP. I have installed it on many a laptop now and it just doesn't seem to function as the solution guide shows.

I've tried this on my caddy 2013 and polo gti 2016. the options are just greyed out. I can find no real guide on how to use vcp so everything is just trial and error.

I load the VCP. I connect to car. (car running/car ignition on) I select more. type manually 5F and it opens up the 5F panel. now this screen is just wrong from any guide I see. the adaption button is greyed out for one.I can select EOL from the drop down it says changed session. but nothing happens and nothing changes. I have tried every sequence I can think of to get this functioning but nothing. none of the menus are populated with options etc.

Sec Access. just plain refuses any login.

I am literally going mad trying to figure this out.

"ASAM nOK" ? I have a feeling this maybe related but I cannot find any info how to fix or understand this issue.

any help would be much appreciated.

all I want to do is enable performance monitor. 

HNY


![IMG_0851](https://user-images.githubusercontent.com/65807296/103470243-a9ec7080-4dbb-11eb-8c03-e6535a6feacb.jpg)

**Comment by alexprojectGH** [comment]
> > Here is not really the place for all these questions, VCRN can be found in the enabled codes menu by selecting one of the codes.
> > Have you got the mega link? You can find it easily online on mhh or CT forums and it answers all of this.
> 
> * [x]  I don't have third option in menu where FEC codes should be, only update and hmi tails. And yeah, I got mega links, updated my unit, patched swap.
> * [x]  I ask so many questions because I really don't want to mess up anything, and also that info will 100% be useful for someone else.
> * [x]  I'm stuck with signed_exception_list.txt and what's content of that file
> * [x]  @mattcabb can you give me usable example of signed_exception_list.txt please?
> 
> Edit:Passing signed_exception_list.txt with just signatures and no FEC codes activates all codes available. Problem solved, CarPlay activated

@fikaa73 Did you unlock the CarPlay? Can you help do that?

**Comment by fikaa73** [comment]
Hi, yes, give me your email please


Srdačan pozdrav,
Filip Vukotić
FIST
20/106i
________________________________

**Comment by alexprojectGH** [comment]
> Hi, yes, give me your email please

My email: alex.tern92@gmail.com

**Comment by jado810301** [comment]
Hello 

What do i need to create the Swap codes???  do anyone teach me or give a right direction to find it?? I would like to upload thios using ODIS, fpor Delphi modules.... thanks

**Comment by fikaa73** [comment]
> Hello
> 
> What do i need to create the Swap codes??? do anyone teach me or give a right direction to find it?? I would like to upload thios using ODIS, fpor Delphi modules.... thanks

I think delphi uses modified feccontainer to activate fecs, but correct me if i'm wrong
You need to extract it from unit, edit it on pc using whatthefec and upload it again

**Comment by cuilh1016** [comment]
@dnoermann hello，my unit HW is 0359. According to what you said, I have modified these two places. The IP of my computer is set to 192.168.1.100, and green menu is also opened with ODIS. However, telnet can't connect to my computer. Where else do I need to set or modify? Thank you very much
 ![pf conf](https://user-images.githubusercontent.com/57856540/108009836-7baab380-703e-11eb-95bc-afeaee2f5e8f.png)
![start](https://user-images.githubusercontent.com/57856540/108009843-7ea5a400-703e-11eb-90eb-14438e888b5a.png)

**Comment by dnoermann** [comment]
@cuilh1016 You need to modify startup to start telnetd

on -p 25 waitfor /net/J5
sleep 0.5

echo /net/J5/dev/ser1 \"/bin/login -f root\" qansi-m on > /tmp/ttys
/sbin/tinit -f /tmp/ttys &
/usr/sbin/telnetd -debug 23 &

**Comment by osiak809** [comment]
part, my MIB2 unit stood on the update screen, does anyone have an idea to enter safe mode without a UART adapter 
![received_4343329822379522](https://user-images.githubusercontent.com/89030980/129589696-cdc4c43a-80b3-4cd9-8859-e2ae1c415fb6.jpeg)

**Comment by osiak809** [comment]
I have obd eleven, but when I try to enter software updates in the basic settings, it pops up that the boundary conditions have not been met

**Comment by danededane** [comment]
> On the ethernet card of the PC.
> 
> I don't think it will go into emergency mode as it is in a constant boot loop.

Did you find a solution on this constant looping device ? I also have a ZR/Technisat that are looping.

**Comment by osiak809** [comment]
Yes

**Comment by osiak809** [comment]
i bought a usb UART and installed the firmware via emergency mode

**Comment by danededane** [comment]
> i bought a usb UART and installed the firmware via emergency mode

Do you remember which MIB version you had?

**Comment by osiak809** [comment]
0245T I installed version 467 and wifi threw me I wanted to install the wifi driver and that's why it stopped and I had to go back to 245T

**Comment by danededane** [comment]
> 0245T I installed version 467 and wifi threw me I wanted to install the wifi driver and that's why it stopped and I had to go back to 245T

Can you remember which pins you used and which usb device for the serial connection?
I'm having a little trouble with som strangly looking outputs even though I've set the baud rate to 115200.
